From a0aca52f6f7b45ac1cbd83e84baf5bcc0a649176 Mon Sep 17 00:00:00 2001
From: lyintel <yang.a.lu@intel.com>
Date: Tue, 31 May 2022 20:28:06 +0000
Subject: [PATCH] INTERNAL: Do not cache depth and stencil

On Android, there will be garbage when screen rotation
during video playback.
This is to revert d245d7b

Signed-off-by: Chenglei Ren <chenglei.ren@intel.com>
Tracked-On: OAM-99738
---
 src/gallium/winsys/virgl/drm/virgl_drm_winsys.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/src/gallium/winsys/virgl/drm/virgl_drm_winsys.c b/src/gallium/winsys/virgl/drm/virgl_drm_winsys.c
index f2336be6fa7..c4e6263fdba 100644
--- a/src/gallium/winsys/virgl/drm/virgl_drm_winsys.c
+++ b/src/gallium/winsys/virgl/drm/virgl_drm_winsys.c
@@ -65,10 +65,14 @@ static inline boolean can_cache_resource(uint32_t bind)
           bind == VIRGL_BIND_INDEX_BUFFER ||
           bind == VIRGL_BIND_VERTEX_BUFFER ||
           bind == VIRGL_BIND_CUSTOM ||
+#if defined(ANDROID)
+          bind == VIRGL_BIND_STAGING;
+#else
           bind == VIRGL_BIND_STAGING ||
           bind == VIRGL_BIND_DEPTH_STENCIL ||
           bind == VIRGL_BIND_RENDER_TARGET ||
           bind == 0;
+#endif
 }
 
 static void virgl_hw_res_destroy(struct virgl_drm_winsys *qdws,
-- 
2.40.1

